controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java

```diff
@@ -683,12 +683,14 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                     }
                 }
             } else {
-                //PathAddress target = aliasEntry.getTargetAddress();
                 PathAddress target = aliasEntry.convertToTargetAddress(addr.append(element));
                 PathAddress targetParent = target.subAddress(0, target.size() - 1);
                 Resource parentResource = context.readResourceFromRoot(targetParent, false);
-                if (parentResource != null && parentResource.hasChildren(target.getLastElement().getKey())) {
-                    set.add(element.getValue());
+                if (parentResource != null) {
+                    PathElement targetElement = target.getLastElement();
+                    if (targetElement.isWildcard() ? parentResource.hasChildren(targetElement.getKey()) : parentResource.hasChild(targetElement)) {
+                        set.add(element.getValue());
```

Comment:
If targetElement.isWildcard(), then the names of all children of that type need to be added. This will add "*'. (The old code did too.) The block starting here does this:

https://github.com/wildfly/wildfly-core/blob/master/controller/src/main/java/org/jboss/as/controller/operations/global/GlobalOperationHandlers.java#L678

This code could do the same thing, but using 'parentResource' instead of 'resource' and 'targetElement.getKey()' instead of 'childType'.